Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/AssetFolderThumbnailView.cpp

```diff
@@ -156,11 +156,20 @@ namespace AzQtComponents
 
         // pixmap
 
-        const auto& iconVariant = index.data(Qt::DecorationRole);
-        if (!iconVariant.isNull())
+        const auto& qVariant = index.data(Qt::DecorationRole);
+        if (!qVariant.isNull())
         {
-            const auto& icon = iconVariant.value<QIcon>();
-            icon.paint(painter, imageRect);
+            QIcon icon;
+            if (const auto& path = qVariant.value<QString>(); !path.isEmpty())
+            {
+                icon.addFile(path, rect.size(), QIcon::Normal, QIcon::Off);
```

Comment:
Will the file path be guaranteed to exist? If not, we may want to check file existence as well

Reply:
If the path doesn't exist then no icon will be painted on the thumbnail, maybe we could print a warning if there is no file?